packages-ts/gauntlet-terra/src/commands/logger.ts

```diff
@@ -0,0 +1,101 @@
+  `[${logger.style(label, ...styles.MULTISIG_LABEL)}🧳${logger.style(address, ...styles.MULTISIG_ADDRESS)}]`
+
+const formatContract = (address: string, label: string): string =>
+  `[👝${logger.style(label, ...styles.CONTRACT_LABEL)}📜$${logger.style(address, ...styles.CONTRACT_ADDRESS)}]`
+
+const formatOperator = (address: string): string =>
+  `[${logger.style('operator', ...styles.OPERATOR_LABEL)}${astronaut}${logger.style(
+    address,
+    ...styles.OPERATOR_ADDRESS,
+  )}]`
+
+export class TerraLogger {
+  addressBook: AddressBook
+
+  withAddressBook(addressBook: AddressBook) {
+    this.addressBook = addressBook
+  }
+
+  // logger.styleAddress():  Format a terra addresses depending on contract type.
+  // Usage:
+  //
+  // 1. Call logger.withAddressBook(addressBook) in middleware
+  //
+  // 2. import { logger } from '@gauntlet-terra/dist/commands/logger'
+  //
+  // 3. Use ${logger.styleAddress(address)} instead of ${address} in strings sent to console or log.
+  //   - If it matches the address added with name='multisig', the address will show up
+  //     as yellow and labelled "multisig".
+  //   - If it matches a known contract address read from the environemnt (LINK, BILLING_ACCESS_CONTROLLER,... ),
+  //     the address will be blue and labelled with the contract name( or "name" if specified )
+  //   - Unknown addresses will remain unformmated
+  styleAddress(address: string): string {
+    if (!this.addressBook) {
+      logger.warn(`TerraLogger: styleAddress called before calling withAddressBook!`)
+      return address
+    }
+
+    if (this.addressBook.instances.has(address)) {
+      const name = this.addressBook.instances.get(address).name
+      if (name == 'multisig') {
+        return formatMultisig(address, name)
+      } else {
+        return formatContract(address, name)
+      }
+    } else if (address == this.addressBook.operator) {
+      return formatOperator(address)
+    } else {
+      return address
+    }
+  }
+}
+
+// TODO:
+// Ideally, the instantiation of all objects should be in terra-gauntlet-contracts,
+//  but moving it there would not allow other packages such as gauntlet-terra-cwplus
+//  to import it directly to use for stylizing addresses... which is important for
+//  multisig-wrapped commands.  We should look for a clean way to be able to call
+//  logger.styleAddress() from gauntlet-terra-cwplus without using "import { logger }"
+//  from there. (Presumably, it will need to call styleAddress() on some other object
+//  that's injected to it from terra-gauntlet-contracts, which will call the instantiated
+//  TerraLogger)
+const terraLogger = new TerraLogger()
+
+export default {
+  table: logger.table,
+  log: logger.log,
+  info: logger.info,
+  warn: logger.warn,
+  success: logger.success,
+  error: logger.error,
+  loading: logger.loading,
+  line: logger.line,
+  debug: logger.debug,
+  time: logger.time,
+  style: logger.style,
```

Comment:
If logger grows we'd lose that new properties:
```suggestion
  ...logger,
```